flink/v1.17/flink/src/main/java/org/apache/iceberg/flink/FlinkCatalog.java

```diff
@@ -472,11 +479,13 @@ public void alterTable(ObjectPath tablePath, CatalogBaseTable newTable, boolean
 
     CatalogTable table = toCatalogTable(icebergTable);
 
-    // Currently, Flink SQL only support altering table properties.
+    // This alterTable API only supports altering table properties.
```

Comment:
Should we deprecate this API?

Reply:
It seems that flink is not deprecating the old alterTable API yet. This is to support some legacy operations: `AlterTableSchemaOperation`, `AlterViewPropertiesOperation` and `AlterViewAsOperation`. 


Among these operations:

- `AlterTableSchemaOperation` is now [only used to alter hive tables](https://github.com/apache/flink/blob/553c72a894b25c10e71e54e6567c187876234fd5/flink-table/flink-table-planner/src/main/java/org/apache/flink/table/planner/utils/OperationConverterUtils.java#L133). 
- `AlterViewPropertiesOperation` and `AlterViewAsOperation` are for views.

So I guess it's safe to assume the old alterTable API won't be called by flink when altering iceberg tables.

Reply:
We cannot deprecate this API from iceberg's side, since it's defined in flink repo. But we can log some warnings inside this method instead. Is that ok?

Reply:
That makes sense to me.